parsl/dataflow/dflow.py

```diff
@@ -296,6 +297,109 @@ def std_spec_to_name(name, spec):
+        return file_log_info
+
+    def register_as_input(self, f: Union[File, DataFuture],
+                          task_record: TaskRecord):
+        """ Register a file as an input to a task. """
+        if self.monitoring and self.monitoring.capture_file_provenance:
+            self._send_file_log_info(f, task_record, False)
+            file_input_info = self._create_file_io_info(f, task_record)
+            self.monitoring.send((MessageType.INPUT_FILE, file_input_info))
+
+    def register_as_output(self, f: Union[File, DataFuture],
+                           task_record: TaskRecord):
+        """ Register a file as an output of a task. """
+        if self.monitoring and self.monitoring.capture_file_provenance:
+            self._send_file_log_info(f, task_record, True)
+            file_output_info = self._create_file_io_info(f, task_record)
+            self.monitoring.send((MessageType.OUTPUT_FILE, file_output_info))
+
+    def _create_file_io_info(self, file: Union[File, DataFuture],
+                             task_record: TaskRecord) -> Dict[str, Any]:
+        """
+        Create the dictionary that will be included in the monitoring db
+        """
+        file_io_info = {'file_id': str(file.uuid),
+                        'run_id': self.run_id,
+                        'task_id': task_record['id'],
+                        'try_id': task_record['try_id'],
+                        }
+        return file_io_info
+
+    def _register_env(self, environ: ParslExecutor) -> None:
+        """ Capture the environment information for the monitoring db. """
+        if self.monitoring and self.monitoring.capture_file_provenance:
+            environ_info = self._create_env_log_info(environ)
+            self.monitoring.send((MessageType.ENVIRONMENT_INFO, environ_info))
+
+    def _create_env_log_info(self, environ: ParslExecutor) -> Dict[str, Any]:
+        """
+        Create the dictionary that will be included in the monitoring db
+        """
+        env_log_info = {'run_id': environ.run_id,
+                        'environment_id': str(environ.uu_id),
+                        'label': environ.label
+                        }
+
+        env_log_info['address'] = getattr(environ, 'address', None)
+        provider = getattr(environ, 'provider', None)
+        if provider is not None:
+            env_log_info['provider'] = provider.label
+            env_log_info['launcher'] = type(getattr(provider, 'launcher', None))
+            env_log_info['worker_init'] = getattr(provider, 'worker_init', None)
+        return env_log_info
+
+    def log_info(self, msg: str) -> None:
+        """Log an info message to the monitoring db."""
+        if self.monitoring:
+            if self.monitoring.capture_file_provenance:
+                misc_msg = self._create_misc_log_info(msg)
+                if misc_msg is None:
+                    logger.info("Could not turn message into a str, so not sending message to monitoring db")
```

Comment:
despite the log message, looks like the code does the send anyway?

This is a user facing call and the user did something wrong with the parameter - this code should stick to Python's normal exception handling, which is to propagate the exception upwards to the user, unless there is a specific reason to behave differently.

Reply:
yep, fixed.